File: app/src/main/java/com/luffy/mulmedia/opengl/VideoShader.java
package com.luffy.mulmedia.opengl;

import android.content.Context;
import android.text.TextUtils;

import com.luffy.mulmedia.utils.FileUtils;

public class VideoShader implements IGLShader {
    private Context context;

    private String vertexShader = null;
    private String fragShader = null;

    public VideoShader(Context context) {
        this.context = context;
    }

    @Override
    public String vertexShader() {
        if (TextUtils.isEmpty(vertexShader)) {
            return FileUtils.getStringFromAssets(context, "shader_vertex_video.glsl");
        }
        return vertexShader;
    }

    @Override
    public String fragmentShader() {
        if (TextUtils.isEmpty(fragShader)) {
            fragShader = FileUtils.getStringFromAssets(context, "shader_frag_video.glsl");
        }
        return fragShader;
    }
}

File: app/src/main/kotlin/com/luffyxu/mulmedia/gles3/video/VideoShader.kt
package com.luffyxu.mulmedia.gles3.video

import android.content.Context
import android.text.TextUtils
import com.luffyxu.opengles.base.shader.IGLShader
import com.luffyxu.opengles.base.utils.FileUtils

class VideoShader(
    var context: Context,
    val vertexShader: String? = null,
    val fragShader: String? = null
) : IGLShader {
    override fun vertexShader(): String {
        return if (TextUtils.isEmpty(vertexShader)) {
            FileUtils.getStringFromAssets(context, "gles3/shader_vertex_video.glsl")
        } else vertexShader!!
    }

    override fun fragmentShader(): String {
        return if (TextUtils.isEmpty(fragShader)) {
            FileUtils.getStringFromAssets(context, "gles3/shader_frag_video.glsl")
        } else fragShader!!
    }
}